e2e/docker/docker.go

```diff
@@ -271,6 +271,36 @@ func (c ctx) testDockerHost(t *testing.T) {
 func (c ctx) testDockerCredsPriority(t *testing.T) {
 	e2e.EnsureImage(t, c.env)
 
+	privImgNoPrefix := strings.TrimPrefix(c.env.TestRegistryPrivImage, "docker://")
+	simpleDef := e2e.PrepareDefFile(e2e.DefFileDetails{
+		Bootstrap: "docker",
+		From:      privImgNoPrefix,
+	})
+	t.Cleanup(func() {
+		if !t.Failed() {
+			os.Remove(simpleDef)
+		}
+	})
+
+	tmpdir, tmpdirCleanup := e2e.MakeTempDir(t, c.env.TestDir, "build-auth", "")
+	t.Cleanup(func() {
+		if !t.Failed() {
+			tmpdirCleanup(t)
+		}
+	})
+
+	dockerfile, err := e2e.WriteTempFile(tmpdir, "Dockerfile", fmt.Sprintf(
+		`
+FROM %s
```

Comment:
If you declare this separately as a const and then use it in the `fmt.Sprintf` the code will be cleared to read. It's difficult to match brackets, commas etc. either side of a multi-line string literal.